### DISCUSSION TITLE
Accessing protected endpoint with automated means i.e scripts

### DISCUSSION THREAD

**Discussion by knuurr** [authelia/authelia#6406] [Q&A] [answered]
Hi,

Within my homelab, most of my endpoint are all protected by Authelia SSO.

However, currently I'm working on script which would fetch Traefik api for dynamically scraping routes which are available.

Endpoint just like almost all other I have is protected by SSO. It of course works well when I have already active session i.e. through browser.

What would be the easiest solution to access such endpoints through programmatic means, without human input, **without** making any exceptions in config files?

Say, I can create dedicated user for scripts, no problem, but what's the best solution to my problem. Can I delcare some immutable secret, which would bypass login need each time, that could be passed with each request?

**Comment by james-d-elliott** [comment, marked as answer]
Currently the best solution is to utilize a specific account for this purpose, allow that account 1FA access from a designated IP, then use the basic scheme with the `Proxy-Authorization` header. We are however introducing an OAuth2 mechanism to handle this scenario. See https://github.com/authelia/authelia/pull/5916

**Comment by james-d-elliott** [comment]
See https://www.authelia.com/integration/openid-connect/oauth-2.0-bearer-token-usage/ for a newer method

**Reply by knuurr** [reply]
Will see, thanks a lot!
